9064 bokeh.command typing

w0rdsm1th opened the pull request:
added mypy typing to function inputs and return values for command package. specific module comments and difficulties below

setup.cfg: reference the package for mypy testing

# subcommand/
[x] __init__.py
bokeh.command.subcommands._collect()
# TODO - added an empty string "name" attribute to Subcommand class, the if-check may now fail
if not hasattr(attr, 'name'): continue # excludes abstract bases

Unsure if should initialise with None instead of empty string. Attributes are not really Optional when subclassed
https://mypy.readthedocs.io/en/latest/kinds_of_types.html#optional-types-and-the-none-type
name: Optional[str] = None
help: Optional[str] = None

[X] serve.py
- found that paths are still string at this point and NOT PathLike
- bokeh.util.string.format_docstring fails typechecking, expects an Optional[str] arg and __doc__ is str

[X] svg.py
assumed below selenium executed JS will return bytes and not bytearray
bokeh.command.subcommands.svg.SVG.file_contents()
_SVG_SCRIPT = """
var serialized_svgs = [];
var svgs = document.getElementsByClassName('bk-root')[0].getElementsByTagName("svg");
for (var i = 0; i < svgs.length; i++) {
    var source = (new XMLSerializer()).serializeToString(svgs[i]);
    serialized_svgs.push(source);
};
return serialized_svgs
"""

# command/
[X] bootstrap.py
- bokeh.command.bootstrap.main(): struggling to resolve argv mypy typing. is nested tuple of [(arg_name, arg_dict)] not Sequence[str]
[X] subcommand.py
- added empty class attributes "name" and "help". could impact hasattr() checks elsewhere

[X] util.py
bokeh.command.util.set_single_plot_width_height()
issue of conditionally assigning int native and Int bokeh type to same variable.
could use class properties and getters/setter methods on the Would need to set it on the Plot object: bokeh/models/plots.py:495 https://www.python-course.eu/python3_properties.php


- [X ] issues: fixes #9064 
- [ ] tests added / passed
N/A - mypy tests
- [ ] release document entry (if new feature or API change)
N/A

bryevdv commented:
@w0rdsm1th Thanks for the PR but FYI `landing-2.0` has already been merged, so this and new PRs should be against `master`. I've changed the base, but there are a few conflicts to fix up. Can you rebase locally against master and force push?

w0rdsm1th commented:
@bryevdv ah right, was merged as part of v1.4 release? just rebased, changed name of my fork branch to "w0rdsm1th/master/9064_bokeh_command_typing" and submitted PR to master now

https://github.com/bokeh/bokeh/pull/9390

bryevdv commented:
superseded by #9390
